Gem/Code/Source/Components/Multiplayer/EnergyBallComponent.cpp

```diff
@@ -39,66 +40,41 @@ namespace MultiplayerSample
 
     void EnergyBallComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
+#if AZ_TRAIT_CLIENT
         m_effect = GetExplosionEffect();
-        m_effect.Initialize();
+        m_effect.Initialize(GameEffect::EmitterType::FireAndForget);
 
-#if AZ_TRAIT_CLIENT
-        BallActiveAddEvent(m_ballActiveHandler);
+        AZ::EntityBus::Handler::BusConnect(GetEntityId());
+        if (cl_EnergyBallDebugDraw)
+        {
+            m_debugDrawEvent.Enqueue(AZ::TimeMs{ 0 }, true);
+        }
 #endif
     }
 
     void EnergyBallComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
 #if AZ_TRAIT_CLIENT
-        m_ballActiveHandler.Disconnect();
+        m_effect = {};
+        AZ::EntityBus::Handler::BusDisconnect();
+        m_debugDrawEvent.RemoveFromQueue();
 #endif
     }
 
 #if AZ_TRAIT_CLIENT
-    void EnergyBallComponent::OnBallActiveChanged(bool active)
+    void EnergyBallComponent::OnEntityDeactivated([[maybe_unused]] const AZ::EntityId& entityId)
```

Comment:
So this is essentially a 'pre deactivate' event that gets triggered on the entity bus for each entity that's about to deactivate?

Reply:
Yep, exactly. You can see the flow here:
```
    void Entity::Deactivate()
    {
        AZ_PROFILE_FUNCTION(AzCore);

        AZ::ComponentApplicationRequests* componentApplication = AZ::Interface<AZ::ComponentApplicationRequests>::Get();
        if (componentApplication != nullptr)
        {
            componentApplication->SignalEntityDeactivated(this);
        }
        EntityBus::Event(m_id, &EntityBus::Events::OnEntityDeactivated, m_id);
        EntitySystemBus::Broadcast(&EntitySystemBus::Events::OnEntityDeactivated, m_id);

        AZ_Assert(m_state == State::Active, "Component should be in Active state to be Deactivated!");
        SetState(State::Deactivating);

        for (ComponentArrayType::reverse_iterator it = m_components.rbegin(); it != m_components.rend(); ++it)
        {
            DeactivateComponent(**it);
        }

        m_transform = nullptr;
        SetState(State::Init);
    }
```

A couple different types of "OnEntityDeactivated" events get triggered (one on a per-ID bus, one on a global bus), then the entity is put in a deactivating state, then the components are deactivated, then the entity is back in its init state.

OnEntityDeactivated is the last chance where you can process something while the entity and its components are still considered active.